juju/model.py

```diff
@@ -1177,11 +1225,12 @@ def name(self):
         return self._info.name
 
     @property
-    def info(self):
+    def info(self) -> ModelInfo:
         """Return the cached client.ModelInfo object for this Model.
 
         If Model.get_info() has not been called, this will return None.
         """
+        assert self._info is not None
         return self._info
```

Comment:
[blocker?] Behaviour change. Intended? Update docstring if so:
```
        If Model.get_info() has not been called, this will raise an ...
```
 Otherwise preserve behaviour and have return type be `ModelInfo | None`

Reply:
Ehh one integration test tripped on this:

```
  File "/home/runner/work/python-libjuju/python-libjuju/juju/model.py", line 1233, in info
    assert self._info is not None
AssertionError
```

I need to examine this closer.
Also, I'm beginning to appreciate all the hard work you did to get integration tests up and running again 🙇🏻

Reply:
added explicit ValueError and refactored the code that uses this path.